mbed/mbed.py

```diff
@@ -2123,21 +2164,63 @@ def sync(recursive=True, keep_refs=False, top=True):
         "View the dependency tree of the current program or library."))
 def list_(detailed=False, prefix='', p_path=None, ignore=False):
     repo = Repo.fromrepo()
+    revtags = repo.scm.gettags(repo.rev) if repo.rev else []
+    revstr = ('#'+repo.rev[:12]+(', tags:'+', '.join(revtags[0:2]) if len(revtags) else '')) if repo.rev else ''
 
-    print "%s (%s)" % (prefix + (relpath(p_path, repo.path) if p_path else repo.name), ((repo.url+('#'+str(repo.rev)[:12] if repo.rev else '') if detailed else str(repo.rev)[:12]) or 'no revision'))
+    print "%s (%s)" % (prefix + (relpath(p_path, repo.path) if p_path else repo.name), ((repo.url+('#'+str(repo.rev)[:12] if repo.rev else '') if detailed else revstr) or 'no revision'))
 
     for i, lib in enumerate(sorted(repo.libs, key=lambda l: l.path)):
-        if prefix:
-            nprefix = prefix[:-3] + ('|  ' if prefix[-3] == '|' else '   ')
-        else:
-            nprefix = ''
+        nprefix = (prefix[:-3] + ('|  ' if prefix[-3] == '|' else '   ')) if prefix else ''
         nprefix += '|- ' if i < len(repo.libs)-1 else '`- '
 
         if lib.check_repo(ignore):
             with cd(lib.path):
                 list_(detailed, nprefix, repo.path, ignore=ignore)
 
 
+# Command release for cross-SCM release tags of repositories
+@subcommand('releases',
+    dict(name=['-a', '--all'], dest='detailed', action='store_true', help='Show revision hashes'),
+    dict(name=['-u', '--unstable'], dest='unstable', action='store_true', help='Show unstable releases well, e.g. release candidates, alphas, betas, etc'),
+    dict(name=['-r', '--recursive'], action='store_true', help='Show release tags for all libraries and sub-libraries as well'),
+    help='Show release tags',
+    description=(
+        "Show release tags for the current program or library."))
+def releases_(detailed=False, unstable=False, recursive=False, prefix='', p_path=None):
+    repo = Repo.fromrepo()
+    tags = repo.scm.gettags()
+    revtags = repo.scm.gettags(repo.rev)  if repo.rev else [] # associated tags with current commit
+    revstr = ('#'+repo.rev[:12]+(', tags:'+', '.join(revtags[0:2]) if len(revtags) else '')) if repo.rev else ''
```

Comment:
This is a bit of a formatting nitpick, but I think it helps users more quickly scan for matching revisions.

Currently its setup to print like the following:

```
`- mbed-os (#182bbd51bc8d, tags:latest, mbed-os-5.6.5)
```

What do you think of the following changes?
- Add a space between `tags:` and the list of tags
- Wrap the list of tags with `[]` to help differentiate the comma-separated list from the commit that comes before it

So as an example:

```
`- mbed-os (#182bbd51bc8d, tags: [latest, mbed-os-5.6.5])
```

Reply:
I've added the space after tags. The list wrapped in `[]` seems a lot like taken from the json/javascript world though.

Reply:
Thanks for adding the spaces. We don't have to use `[]`, I was mostly thinking about how we can we show that `latest` and `mbed-os-5.6.5` are part of the `tags` instead of another item in the existing list (the one containing `#182bbd51bc8d` and `tags: ...`). My eyes in particular have trouble separating all the different bits in this line since it contains a lot of (very useful) information.

If you think `()` would be a better container than `[]`, that's totally fine with me! But if you don't think this is necessary, then no biggie!